src/Http/Http/src/Features/FormFeature.cs

```diff
@@ -181,27 +181,29 @@ private async Task<IFormCollection> InnerReadFormAsync(CancellationToken cancell
                         {
                             var fileSection = new FileMultipartSection(section, contentDisposition);
 
-                            // Enable buffering for the file if not already done for the full body
-                            section.EnableRewind(
-                                _request.HttpContext.Response.RegisterForDispose,
-                                _options.MemoryBufferThreshold, _options.MultipartBodyLengthLimit);
-
-                            // Find the end
-                            await section.Body.DrainAsync(cancellationToken);
-
                             var name = fileSection.Name;
                             var fileName = fileSection.FileName;
 
                             FormFile file;
                             if (section.BaseStreamOffset.HasValue)
```

Comment:
I expect we'll need to re-think what buffering means in the pipe scenario. This concept of base offsets assumed the underlying request body implementation was a seekable stream, which doesn't make sense here anymore.

One idea would be to create a FileBufferingPipeReader implementation similar to FileBufferingReadStream, wrap that around the pipe you want to buffer (the multipart section pipe), and call a BufferAsync(CT) method on it that drained the underlying pipe into a buffer. Not sure if you'd need a concept of Seek here, but if the underlying representation is a file stream then that might still be possible. How does this sound?

Reply:
I guess this isn't for me to decide, but if I can weigh in, I think that this should be a part of another PR. 
If I understand correctly the benefit you are trying to gain is to allow developers be able to get a PipeReader and not just a Stream from FormFile.  This seems like a welcome change - however it shouldn't block this PR, which delivers a way to the BodyReader to parse a MultipartFormFeature (with performance benefits).

Reply:
There are two parts to this concern.
A) Exposing pipes end-to-end.
B) Making sure the existing scenarios still work as expected.

A can wait for another PR only if B can be made to work without it. The current design doesn't fully account for the buffered scenarios. We need to get those worked out before merging.

Reply:
continuing this conversation on the FormFile comment thread